tests/jballerina-unit-test/src/test/resources/test-src/types/xml/xml-indexed-access.bal

```diff
@@ -88,6 +88,362 @@ function testLengthOfXMLSequence() returns [int, int, int, int] {
+    assert((<error> e).message(), errorMessage);
+    e = trap x8[a];
+    assertTrue(e is error);
+    assert((<error> e).message(), errorMessage);
+    e = trap x8[i];
+    assertTrue(e is error);
+    assert((<error> e).message(), errorMessage);
+
+    xml x9 = xml `<names><name>Sherlock Holmes</name><name>Sherlock Holmes</name></names>`;
+    e = trap x9[1][-1];
+    assertTrue(e is error);
+    assert((<error> e).message(), errorMessage2);
+    e = trap x9[1][a];
+    assertTrue(e is error);
+    assert((<error> e).message(), errorMessage2);
+    e = trap x9[1][i];
+    assertTrue(e is error);
+    assert((<error> e).message(), errorMessage2);
+
+    e = trap x9[0][-1];
+    assertTrue(e is error);
+    assert((<error> e).message(), errorMessage);
+    e = trap x9[0][a];
+    assertTrue(e is error);
+    assert((<error> e).message(), errorMessage);
+    e = trap x9[0][i];
+    assertTrue(e is error);
+    assert((<error> e).message(), errorMessage);
+
+
+    xml x10 = xml `<!--This is a comment 1--><!--This is a comment 1-->`;
+    e = trap x10[1][-1];
+    assertTrue(e is error);
+    assert((<error> e).message(), errorMessage);
+    e = trap x10[1][a];
+    assertTrue(e is error);
+    assert((<error> e).message(), errorMessage);
+    e = trap x10[1][i];
+    assertTrue(e is error);
+    assert((<error> e).message(), errorMessage);
+
+    xml x11 = xml `<?target data?><?target data?>`;
+    e = trap x11[1][-1];
+    assertTrue(e is error);
+    assert((<error> e).message(), errorMessage);
+    e = trap x11[1][a];
+    assertTrue(e is error);
+    assert((<error> e).message(), errorMessage);
+    e = trap x11[1][i];
+    assertTrue(e is error);
+    assert((<error> e).message(), errorMessage);
+
+    xml x12 = xml `<names><name>Sherlock Holmes</name><name>Sherlock Holmes</name></names>`;
+    e = trap x12[-1];
+    assertTrue(e is error);
+    assert((<error> e).message(), errorMessage);
+    e = trap x12[a];
+    assertTrue(e is error);
+    assert((<error> e).message(), errorMessage);
+    e = trap x12[i];
+    assertTrue(e is error);
+    assert((<error> e).message(), errorMessage);
+
+    xml x13 = xml `<!--This is a comment 1--><!--This is a comment 1-->`;
+    e = trap x13[-1];
+    assertTrue(e is error);
+    assert((<error> e).message(), errorMessage3);
+    e = trap x13[a];
+    assertTrue(e is error);
+    assert((<error> e).message(), errorMessage3);
+    e = trap x13[i];
+    assertTrue(e is error);
+    assert((<error> e).message(), errorMessage3);
+
+    xml x14 = xml `<?target data?><?target data?>`;
+    e = trap x14[-1];
+    assertTrue(e is error);
+    assert((<error> e).message(), errorMessage3);
+    e = trap x14[a];
+    assertTrue(e is error);
+    assert((<error> e).message(), errorMessage3);
+    e = trap x14[i];
+    assertTrue(e is error);
+    assert((<error> e).message(), errorMessage3);
+
+    xml x15 = xml `<name>Sherlock Holmes</name><name>Sherlock Holmes</name>`;
+    e = trap x15[-1];
```

Comment:
Better add test cased for `-2`, `-3` as well.

Reply:
the check is `<0`, because of that isn't it ok to add only one negative number.
wdyt?